lib/codecs/src/encoding/format/gelf.rs

```diff
@@ -0,0 +1,350 @@
+                if key == VERSION
+                    || key == HOST
+                    || key == SHORT_MESSAGE
+                    || key == FULL_MESSAGE
+                    || key == FACILITY
+                    || key == FILE
+                {
+                    if !value.is_bytes() {
+                        return Err(
+                            format!("LogEvent field '{}' should be a UTF-8 string", key).into()
+                        );
+                    }
+                }
+                // validate timestamp value
+                else if key == TIMESTAMP {
+                    if !(value.is_timestamp() || value.is_integer()) {
+                        return Err(format!(
+                            "LogEvent field '{}' should be a timestamp type or integer",
+                            log_schema().timestamp_key()
+                        )
+                        .into());
+                    }
+                }
+                // validate integer values
+                else if key == LEVEL {
+                    if !value.is_integer() {
+                        return Err(format!("LogEvent field {} should be an integer", key).into());
+                    }
+                }
+                // validate float values
+                else if key == LINE {
+                    if !(value.is_float() || value.is_integer()) {
+                        return Err(format!("LogEvent field '{}' should be a number", key).into());
+                    }
+                } else {
+                    // Additional fields must be prefixed with underscores.
+                    // Prepending the underscore since vector adds fields such as 'source_type'
+                    // which would otherwise throw errors.
+                    if !key.is_empty() && !key.starts_with('_') {
+                        if self.conformed_log.is_none() {
+                            self.conformed_log = Some(log.clone())
+                        }
+                        self.conformed_log
+                            .as_mut()
+                            .unwrap()
+                            .rename_key(key.as_str(), &*format!("_{}", &key));
+                    }
+
+                    // additional fields must be only word chars, dashes and periods.
+                    if !self.valid_regex.is_match(key) {
+                        return Err(format!(
+                            "LogEvent field '{}' contains an invalid character",
+                            key
+                        )
+                        .into());
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    /// Determine if input log event is in valid GELF format. Possible return values:
+    ///    - Ok(None)           => The log event is valid GELF
+    ///    - Ok(Some(LogEvent)) => The the log event isn't stricly valid GELF, but was
+    ///                            conformed to be valid.
```

Comment:
Is this still used? Or is it considered "conformed" when we prepend `_`

Reply:
good catch- before storing conformed in the struct, I was returning it. Need to update that comment.